libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1565,6 +1565,16 @@ void TypeChecker::endVisit(ElementaryTypeNameExpression const& _expr)
 
 void TypeChecker::endVisit(Literal const& _literal)
 {
+	if (_literal.looksLikeAddress())
+	{
+		if (_literal.passesAddressChecksum())
```

Comment:
Shouldn't we check here for length and abort early with a different message?

I think since we're not even checking the characters if it is too short, it would make more sense to return false in `looksLikeAddress` if it is not of correct length.